src/clients/c/tb_client/signal.zig

```diff
@@ -86,7 +99,9 @@ pub const Signal = struct {
     }
 
     fn wait(self: *Signal) void {
-        assert(!self.stop_requested());
+        // It is not guaranteed to be `running` here, as another caller might have requested
+        // a stop during the callback.
+        assert(self.status() != .shutdown);
```

Comment:
Let's cover this scenario by a test. Rather than adding a specific test here, I think we can change our `test "signal"` to be a fuzz-test of sorts --- run it ten times, where sometimes we randomly stop the signal?